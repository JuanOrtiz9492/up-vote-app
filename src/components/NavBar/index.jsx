import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { constants } from '../../config/constants';
import { cssConstants } from '../../config/cssConstants';
import { Link } from "react-router-dom";


const StyledHeaderContainer = styled.header`
    width: ${constants.hero.background.width};
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
`
const StyledNavBar = styled.nav`
    display: inline-flex;
    flex-direction: row;
    width: max-content;
    margin-right: 4em;
    line-height: 2em;
`
const StyledTitle = styled.h1`
    margin-left: 4em;
    width: max-content;
    display: inline;
    color: ${cssConstants.colors.white.hex}
`
const StyledLink = styled(Link)`
    text-decoration:none;
    margin: 1em;
    color: ${cssConstants.colors.white.hex}
`
const NavBar = ({title, linkArray}) => (
    <StyledHeaderContainer>
        <StyledTitle>
            {title}
        </StyledTitle>
        <StyledNavBar>
            {linkArray.map( item => (
                <StyledLink 
                    key={item.id} 
                    to={item.route}
                >
                    {item.children}
                </StyledLink>
            ))}
        </StyledNavBar>
    </StyledHeaderContainer>
)

NavBar.propTypes = {
    title: PropTypes.string,
    linkArray: PropTypes.array,
};

NavBar.defaultProps = {
    title: '',
    linkArray: [],
}

export default NavBar;