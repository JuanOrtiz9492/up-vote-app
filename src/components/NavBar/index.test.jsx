import React from 'react';
import { render } from '@testing-library/react';
import NavBar from '.';

describe('NavBar Component', () => {

    it('should Match the Snapshot', () => {
        const { container } = render(<NavBar />);
        expect(container).toMatchSnapshot();
      })
})

