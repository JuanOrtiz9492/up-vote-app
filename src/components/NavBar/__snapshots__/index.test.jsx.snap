// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component should Match the Snapshot 1`] = `
<div>
  <header
    class="sc-AykKI fgxbaB"
  >
    <h1
      class="sc-AykKK RakPd"
    />
    <nav
      class="sc-AykKJ chLIen"
    />
  </header>
</div>
`;
